replica/database.hh

```diff
@@ -932,7 +935,7 @@ public:
     }
 
     const db::view::stats& get_view_stats() const {
-        return _view_stats;
+        return *_view_stats;
```

Comment:
why change the stats?
the only thing you need to do is delete (like you did ) the _metrics member

Reply:
if I only delete `_metrics` and I don't remove `_view_stats`, I get the same `double_registration` exception. Here's an excerpt from logs (with my additional logging and comments), where you can see that view's metrics still block creation of another table with the same name - view's metric of the removed table still exist while the new table is created and are only removed later:
```
// both tables exist as objects, but the 1st one has _metrics deleted
// second table tries to register all its metris/stats
ERROR 2023-05-18 11:02:56,278 [shard 0] raft - [623619db-3c34-4b7e-931e-08594b085852] applier fiber stopped because of the error: std::_Nested_exception<raft::state_machine_error> (State machine error at raft/raft.hh:299): seastar::metrics::double_registration (registering metrics twice for metrics: column_family_view_updates_pushed_remote)
// frist table's view's stats are deregistered
~metric_groups_impl()
unregister_metric(i) column_family_view_updates_pushed_remote
unregister_metric(i) column_family_view_updates_failed_remote
unregister_metric(i) column_family_view_updates_pushed_local
unregister_metric(i) column_family_view_updates_failed_local
unregister_metric(i) column_family_view_updates_pending
```
So it's not enough to remove only metrics, and when I remove view's stats, the test passes. But I wonder if I shouldn't also remove remaining table's stats/metrics, not only the ones that are causing the problem. Metrics/stats I can see in the `table` class are:
```
class table : public enable_lw_shared_from_this<table> {
...
    replica::cf_stats* cf_stats = nullptr;
...
    mutable table_stats _stats;
    mutable std::unique_ptr<db::view::stats> _view_stats;
    mutable row_locker::stats _row_locker_stats;
...
    std::unique_ptr<seastar::metrics::metric_groups> _metrics = std::make_unique<seastar::metrics::metric_groups>();
```
I currently only delete 2 out of 5.

Reply:
I think this is still unsafe, you need to delete just the _metrics member from write_stats not the entire stats object

Reply:
@amnonh thanks, you're right. I did as you told and everything works.